### PR TITLE
Support "excluded" param in Filtersearch request

### PR DESCRIPTION
**Pull request by yen-tt**
this pr adds support for `excluded` param for FilterSearch request following [liveAPI docs](https://hitchhikers.yext.com/docs/liveapis/answersapi/verticalsearch/#operation/filtersearch). The param is a URL-encoded string containing a JSON array containing the IDs of all filters that should be excluded from filter search results.

Note: Watson identified an edge case of the current implementation with this param - excluded place filters cause all filters to be removed.

J=SLAP-1077
TEST=manual&auto

see that the updated jest tests passed
see that filter search request from test-site, without excluded params, contains `tom dixon` and `tom elliot`. After adding `excluded` param to the request. See that the same set of results were returned but without `tom dixon` and `tom elliot`.

### CODE REVIEW

**Review by oshi97 — APPROVED** [review]

File: src/infra/AutocompleteServiceImpl.ts [hunk 1]
```diff
@@ -177,4 +180,14 @@ export class AutocompleteServiceImpl implements AutocompleteService {
       }
     ));
   }
+
+  private transformExcludedFields(excludedFields?: Filter[]) {
```
[the remark this answers is not in the document]
Comment (oshi97) [reply src/infra/AutocompleteServiceImpl.ts:184]: or I guess this is for a request so [serializers](https://github.com/yext/answers-core/tree/4fdc6ed650957f7f8d244ca16d76564c2a162099/src/serializers)?

[the remark this answers is not in the document]
Comment (yen-tt) [reply src/infra/AutocompleteServiceImpl.ts]: sure! moved this function and the one above to a file in serializer folder

File: src/infra/AutocompleteServiceImpl.ts [hunk 2]
```diff
@@ -167,14 +172,4 @@ export class AutocompleteServiceImpl implements AutocompleteService {
 
     return createFilterSearchResponse(response);
   }
-
-  private serializeSearchParameterFields(fields: SearchParameterField[]) {
```
Comment (oshi97) [inline src/infra/AutocompleteServiceImpl.ts:171]: looks like this was a remnant from the original implementation 😅

Comment (oshi97) [reply src/infra/AutocompleteServiceImpl.ts:171]: https://github.com/yext/answers-core/pull/21

**Review by oshi97 — APPROVED** [review]
